22.md

```diff
@@ -0,0 +1,40 @@
+NIP-22
+======
+
+Connection Limitations
+----------------------
+
+`draft` `optional`
+
+Upon connection, and at any point during the connection, the Relay can send a `LIMITS` payload to the Client reflecting the current state of the user's rights in the relay. 
+
+```json
+["LIMITS", { <limit_properties> }]
+``` 
+
+Where `limit_properties` includes properties that are specifically designed to change the way Clients communicate. Options are: 
+
+- `can_write: <true|false>`, If false, Clients MUST NOT publish events to the relay. 
+- `can_read: <true|false>`, If false, Clients MUST NOT send `REQ` commands to the relay.
+<br/><br/>
+- `accepted_event_kinds: [<unsigned int>, <unsigned int>, ...]`, Clients MUST filter publishing events by these types.
+- `blocked_event_kinds: [<unsigned int>, <unsigned int>, ...]`, Clients MUST remove these types from publishing actions.
+<br/><br/>
+- `min_pow_difficulty: <int>`: Clients MUST filter publishing events if they don't have at least this PoW bit difficulty.
+<br/><br/>
+- `max_message_length: <int>`, Clients MUST filter publishing events and crop/chunk `REQ` filters by this length.
+- `max_subscriptions: <int>`, Clients MUST reduce the number of subscriptions to this amount.
+- `max_filters: <int>`, Clients MUST crop/chunk `REQ` filters by this amount.
+- `max_limit: <int>`, Clients MUST clump each filter's `limit` value to this number and use `since` and `until` to get additional results.
+- `max_event_tags: <int>`, Clients MUST filter publishing events by this `.tags` length.
+- `max_content_length: <int>`, Clients MUST filter publishing events by this `.content` length.
+<br/><br/>
+- `created_at_lower_limit: <unix timestamp>`, Clients MUST filter publishing events whose `.created_at` is smaller than this number
+- `created_at_upper_limit: <unix timestamp>`, Clients MUST filter publishing events whose `.created_at` is bigger than this number
```

Comment:
Can these two be offsets from current time, not absolute timestamps? If I want to limit events up to 1 minute into the future (to allow for time difference between relay&client) I would have to resend the LIMITS message every minute or even more often

Reply:
ohhh nice!

Reply:
done